### DISCUSSION TITLE
FAQs

### DISCUSSION THREAD

**Discussion by DarwinsBuddy** [DarwinsBuddy/WienerNetzeSmartmeter#19] [Announcements]
I moved this to our wiki. ;)

[FAQs](https://github.com/DarwinsBuddy/WienerNetzeSmartmeter/wiki/FAQs)

**Comment by TheRealVira** [comment]
Hey @DarwinsBuddy , when I integrate this into our home assistant I get a kwh sensor and a hidden cost sensor. The cost sensor is always 0,-€ and the kwh is always 419,32. Both sensors haven't changed for two weeks (since integration). I want to help and fix this issue, but I am not sure where to start. Are there any logs I can check for?

**Reply by DarwinsBuddy** [reply]
hi there

I'd suggest to start looking at the logs of hassio
https://www.home-assistant.io/integrations/system_log/

Regarding the cost sensor:

I am not aware of a cost sensor to be exposed by the integration. double check if you have the newest version of the integration installed. there should be for every "Zaehlpunkt" in your wiener netze contract a sensor exposed by the integration which reports the meter readings in total as returned by the API. double check the Zaehlpunktnr. with the sensor entity (they should coincide)

Get back if you need further help. 🤞

**Reply by fdinhobl** [reply]
Hi @TheRealVira, maybe this is related to the following: WienerNetze is not updating the total "Zählerstand" since 1st of December. While they do provide daily consumption and even 15-minute values on their frontend, the total consumption is not updated for some reason. Just expand the month December, you will find "Keine Daten vorhanden" for "Zählerstand" for every single day except 1st of December, at least that's how it is for me and I'm assuming for you as well.

**Reply by DarwinsBuddy** [reply]
cool finding. wasn't aware of that @reox is working on a PR to get some more robust meter readings into our sensors. 😬

**Reply by DarwinsBuddy** [reply]
can confirm. last meter reading for me was on 03.12.2022

**Reply by reox** [reply]
maybe that is because of the billig period? If you log in to wienenergie they also say that it can come to delays with the billig. but that is just a wild guess...
I test my PR currently myself, and it fetched the values now for 4 weeks without major issues.

**Reply by TheRealVira** [reply]
![Screenshot_20221213-173602_Home Assistant](https://user-images.githubusercontent.com/10862533/207391112-6d7628de-2422-4c5e-aed8-989ebb416b74.png)

Weird.. maybe this cost sensor got auto generated by home assistant after adding it to the Energy dashboard? Also apparently I cannot unhide it lol

@reox is right. Thanks for your support!

**Reply by fdinhobl** [reply]
I assume you are testing with daily consumption values @reox ? Those work indeed, but the total number is not increasing for whatever reason (for December, all other months work). Wild guess here: The governmental program to minimize energy-price-impact has been started 1st of December. WienerNetze is obliged to report consumption I would assume in order to have a governance structure around state-subsidies. This could be causing some delays since they definitely don't want to report wrong numbers.

Guess we won't find out though. Appreciate the activity here and the effort in general.

**Reply by TheRealVira** [reply]
@fdinhobl @reox  Where do you get the daily values from? I only receive the (currently static) total.

**Reply by fdinhobl** [reply]
@TheRealVira 

https://smartmeter-web.wienernetze.at/#/Zaehlerstaende

Just in case you would be referring to the application ... This is not integrated into the prod-version yet, they are working on it :)

**Reply by TheRealVira** [reply]
Ahh okok I thought you meant wnsm integration. Thanks for the clarification @fdinhobl

**Reply by reox** [reply]
> @fdinhobl @reox Where do you get the daily values from? I only receive the (currently static) total.

from this branch: https://github.com/reox/WienerNetzeSmartmeter/tree/statistics

**Reply by reox** [reply]
The cost sensor is very likely autogenerated from homeassistant.

**Reply by TheRealVira** [reply]
Awesome! Looking forward to see daily values in main @reox :3

**Reply by Kuschla** [reply]
> > @fdinhobl @reox Where do you get the daily values from? I only receive the (currently static) total.
> 
> from this branch: https://github.com/reox/WienerNetzeSmartmeter/tree/statistics

Sorry for the newbee question, but how would I "change the integration" to the branch which shows 15min based readings in HACS?

**Reply by TheRealVira** [reply]
FYI @Kuschla you could always just add their repository as a "custom repository".

**Reply by DarwinsBuddy** [reply]
or you simply wait for it to be stable and merged into the main repo :)

**Reply by Kuschla** [reply]
Yeah - I'll be patiently waiting then :)
thx for the cool work to all of you and as well for the progress you've been making over the past couple of weeks!

**Comment by Alfred99-HA** [comment]
First of all many thankx for provind the smartmeter integration Wiener Netze for HA! Is was easy to install even tough I am a newbie. It neede a little more reading to understand why it is statistics, ...
Since I am getting a PV (this was the reason to start with HA) I would like to receive the information also for my "Einspeise-Zählpunkte" as well well.
My question is on whether this is working as well and how to add the other Zählepunkt-Identification

**Reply by reox** [reply]
This should work out of the box, as all zählpunkte should be added. Unfortunately, I cannot test this, as I only have a single smartmeter :/
Does it not add the other zählpunkte?

**Reply by Alfred99-HA** [reply]
Thanks,. My Zählpunkt for Feed-In was unlocked yesterday I added it in config. The sensor is visible now, statistics not yet - I will check tomorrow since today is the first metered feed-in.

**Reply by Alfred99-HA** [reply]
Actually couldn't manage to read both "Zählpunkte" - refer to #51

**Comment by GeraldPape** [comment]
How about two Smartmeters under the same contract. e.g. I have a PV System producing energy measured by the second SM. I see the second SM but it never has any values.

**Comment by nolzi84** [comment]
Hi! First of all, i'm a newbie. :)
In the about Section i can read "A home-assistant integration supporting Netz NÖ smartmeters as sensors".
How can i integrate Netz NÖ smartmeters?

**Reply by W-M-B** [reply]
The wnsm is connecting to the smartmeter portal of Wiener Netze.
Ans Netz Nö Most likely has another system. It will not work.

**Comment by tacsist** [comment]
Hi, thank you very much for the project.

I have just installed the integration and would like to know what is the most basic af setup for HA Energy setup? As far as I understood, with >1.7.0 I should use `sensor.at00...` for "Consumed energy" entry and `wnsm:at00...` for "use an entity tracking total cost"? Is there any other things to enable on new install?

Would be nice to have this in readme or FAQ as well (unless I am dumb and misunderstood completely)

**Reply by tacsist** [reply]
After some tinkering I understood that the sensor is just most recent consumption value and therefore wnsm should be set as "consumed energy". But then I cannot set the price value, and the only option available is "Use an entity tracking the total costs" which seems to report 1eur/kWh with wnsm for now.

**Comment by kaho2005** [comment]
First of all thanks for the great work.
I am trying to integrate 2 smartmeters into HA. One is for TAGSTROM and he other one is for NACHTSTROM.
Under developer tools -> statistics both are visible as wnsm.... and sensor....
But only one sensor.... is showing me statistics. The other one is showing "No Statistics found"
I read above somebody having the same issue. Is there any way to get the statistics for both smartmeters?

**Comment by scw2wi** [comment]
On the website https://smartmeter-web.wienernetze.at/ I can see my data in a 15 min internal, but in HA not.
Is it possible to configure my sensor in HA to get data in HA also in a 15 min interval?

**Reply by Manuel2030** [reply]
that would be awesome, i m thinking also about that because right now you have only the daily consumption..
we need to fetch the full day in 15mins intervals :)

**Comment by krauthaus** [comment]
In my case I got a weird value from that sensor; which is about 1/4 of the actual number on the webpage of the smartmeter.

**Reply by mariomurrent-softwaresolutions** [reply]
> ###

Same here - what are the values actually displaying? The total production of the year? day? whatever? They are confusing

**Reply by W-M-B** [reply]
The plugin collects and display the data from the smartmeter portal.
In my case I get hourly values.
This are then able to get aggregated.
15 minutes intervals I have not seen. I find them also confusing as kWh per 15 minutes is not easy to digest as unit.
<img width="1170" height="1633" alt="IMG_4207" src="https://github.com/user-attachments/assets/41797d75-abaf-4300-a32c-07a2b690c9cb" />

**Comment by Puca89** [comment]
Hi, 
also from my side thanks for the great work. 
wnms hasn't been working for a few days now. So investigated why it not work anymore and I think the reason is the login to the Wiener Netze Smart Meter page. I changed from password to passkey and in my home assitant instance I loged in via password. So I guess I broke wnms with this change. 
So I don't know if there will be a solution cause this not only realted to wnms, so I think this is a general point to home assistant.

**Reply by W-M-B** [reply]
Passkey is pinned to your device/ profile I am not aware that home assistant can use passkeys.

**Comment by Hans1-LB** [comment]
Hey, really great integration! Im really happy with it, just noticed a small thing - did anybody else notice, that the date of the readings is off by 2 days?
I just compared the home assistant readings to the values in the web portal, and the values in home assistant are always 2 days in the future. Is there a way to fix this?
It would be nice, because I also use smart plugs for some devices that measure power consumption, and in the energy tab it displays me overall and device consumption, but it wouldn't match if the days are shifted.

**Comment by namanuel** [comment]
Hey,

i love your integration, but unfortunately, it kinda stopped working a week ago, and I dont know why, as I am concerned i havent changed anything besides doing HA updates.

<img width="1657" height="354" alt="image" src="https://github.com/user-attachments/assets/b86f55da-11eb-4aa2-b79c-463ccca55f23" />

I tried both login methods. Do you know any fixes?

Best regards

**Reply by GeraldPape** [reply]
Not the dev, but whats in the logs?

**Reply by reox** [reply]
it looks like the API has some hickups lately - or they also changed something... Always keep in mind, this is with a reverse-engineered API... There is now an integration (see PR) that uses the official API, but it requires you to request some API keys.

**Reply by mundschenk-at** [reply]
My energy provider hasn't gotten any usage stats since 14.06.2026 either, so it's not just the public API.

**Reply by reox** [reply]
yes, there were also some maintenance announced in the WN-Portal... But even with data available some reported there are issues with the API (See also #357 )

**Comment by zerobit47** [comment]
Hi, first of all, thank you for creating and maintaining this integration! It’s very useful for getting Wiener Netze Smart Meter data into Home Assistant.

I noticed a small issue with the Energy Dashboard:

The consumption values themselves seem to be correct, but they appear to be assigned to the wrong day. For example, Wiener Netze shows about 5.9 kWh for August 17, while Home Assistant shows those 5.9 kWh on August 19.

It looks like the integration updates the cumulative energy sensor when the data becomes available from Wiener Netze (usually 1–2 days later). Home Assistant then attributes the consumption to the day the sensor value was updated, rather than to the day the energy was actually consumed.

Is there a way for the integration to add the historical data to Home Assistant statistics using the original measurement date?

Thanks again for your work on this integration!

**Reply by reox** [reply]
> Home Assistant then attributes the consumption to the day the sensor value was updated, rather than to the day the energy was actually consumed.

If there is no newer API for that, that is exactly how HA works. Values are always associated with the time when they appear, and cannot be back-dated. The only "cheat" that the integration uses is to import statistics with the correct time. This however does not work for values...
